c2corg_api/ext/colander_ext.py

```diff
@@ -53,7 +53,10 @@ def deserialize(self, node, cstruct):
         if cstruct is null or cstruct == '':
             return null
         try:
-            return wkbelement_from_geojson(cstruct, self.srid)
+            geom_type = json.loads(cstruct)['type'].upper()
```

Comment:
This way you are calling `json.loads` two times on `cstruct`: Here and in `wkbelement_from_geojson`. Could you do it so that it is called only once?